scripts/grep_more_ioc.py

```diff
@@ -0,0 +1,511 @@
+
+    print_frame.add_argument('-s', '--print_dirs', action='store_true',
+                             default=False,
+                             help='Prints the child & parent IOC'
+                             + 'directories as the final output')
+# --------------------------------------------------------------------------- #
+# search subarguments
+# --------------------------------------------------------------------------- #
+    search = subparsers.add_parser('search',
+                                   help='For using regex-like searches in the'
+                                   + ' child IOC.cfg captured by grep_ioc.'
+                                   + ' Useful for quickly gathering instance'
+                                   + ' information, IP addr, etc.')
+
+    search.add_argument('search',
+                        type=str, help='PATT to use for regex search in file',
+                        metavar='PATT')
+    search.add_argument('-q', '--quiet', action='store_true', default=False,
+                        help='Surpresses file warning for paths that do not'
+                        + 'exist.')
+    search.add_argument('-o', '--only_search', action='store_true',
+                        default=False,
+                        help="Don't print the dataframe, just search results.")
+    return parser
+
+###############################################################################
+# %% Main
+###############################################################################
+
+
+def main():
+    """
+    Main entry point of the program. For using with CLI tools.
+    """
+    parser = build_parser()
+    args = parser.parse_args()
+    # read grep_ioc output
+    data = find_ioc(args.hutch, args.patt)
+
+    # exit if grep_ioc finds nothing
+    if (data is None or len(data) == 0):
+        print(f'{Fore.RED}No IOCs were found.\nExiting . . .{Style.RESET_ALL}')
+        sys.exit()
+
+    # create the dataframe after fixing the json format
+    df = pd.json_normalize(data)
+
+    # reorder the dataframe if searching all hutches
+    if args.hutch == 'all':
+        df.insert(0, 'hutch', df.pop('hutch'))
+
+    # pad the disable column based on the grep_ioc output
+    if 'disable' not in df.columns:
+        df['disable'] = df.index.size*[False]
+    if 'disable' in df.columns:
+        df.disable.fillna(False, inplace=True)
+
+    # Fill the NaN with empty strings for rarely used keys
+    for _col in df.columns:
+        if _col not in ['delay']:
+            df[_col].fillna('', inplace=True)
+        else:
+            df[_col].fillna(0, inplace=True)
+
+    # check for the ignore_disabled flag
+    if args.ignore_disabled is True:
+        df = df[~df.disable].reset_index(drop=True)
+
+# --------------------------------------------------------------------------- #
+# %%% print
+# --------------------------------------------------------------------------- #
+    # print the dataframe
+    if hasattr(args, 'print'):
```

Comment:
I think the sparse nature of argparse (:badumtiss:) including only the invoked parser & subparser made it weird for testing, and I got distracted by other things and didn't leave a note to clean up.

I should probably [do something with subparser.set_default](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.set_defaults) to give me some namespace clues. Something like `subparser.set_default(command='print')` etc.

Reply:
Turns out my original approach is one of the few stable ways to check for other possible `args`. When `argparse` parses the args, if any subparser is unused, none of its attributes get populated in the resultant `Namespace` object. 
![image](https://github.com/pcdshub/engineering_tools/assets/149725219/a2389126-e7bd-4f14-8812-63274a3a3ace)

I guess the other way of doing this would be generating a full `Namespace` object with all of the defaults set and to check the `bool` of any `sub-commands` because I already structured them as simple toggles at the top level. 
Something like:

```
class Args:
    self.patt = None
    self.hutch = None
    self.ignore_disabled = False
    ...
````

But then I'm hard coding all the defaults anyway that `argparse` was originally handling. 🤷